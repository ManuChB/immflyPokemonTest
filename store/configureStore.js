import {
    createStore,
    combineReducers,
    applyMiddleware
} from 'redux';
import createSagaMiddleware from 'redux-saga';

import pokemonReducer from '../src/reducers/pokemonList.component.reducer';
import rootSaga from '../src/sagas/index';


const rootReducer = combineReducers({
    pokemonList: pokemonReducer
});

const sagaMiddleware = createSagaMiddleware();

const configureStore = () => {
    const store = createStore(
        rootReducer, 
        applyMiddleware(sagaMiddleware) 
    );
    sagaMiddleware.run(rootSaga);
    return store;
};


export default configureStore;