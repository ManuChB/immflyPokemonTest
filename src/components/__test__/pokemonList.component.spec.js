import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import PokemonList from '../pokemonList.component';
import configureStore from '../../../store/configureStore';

configure({ adapter: new Adapter() });

const middlewares = [];
const mockStore = configureStore(middlewares);

describe('PokemonList component', () => {

    it('renders as expected', () => {
        //when
        const wrapper = shallow( 
            <PokemonList store= {mockStore}></PokemonList>
        );
        //expect
        expect(wrapper.dive()).toMatchSnapshot();
    });
});