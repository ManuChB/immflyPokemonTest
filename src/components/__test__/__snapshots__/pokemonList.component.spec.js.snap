// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokemonList component renders as expected 1`] = `ShallowWrapper {}`;
